### DISCUSSION TITLE
Can remix be this usefull and fast if we separate the backend code from it

### DISCUSSION THREAD

**Discussion by emadbaqeri** [remix-run/remix#4121] [Q&A]
Hi folks, I'm really into Remix and thinking about using it in a project of mine after watching Kent Dodds youtube stream.
We are actually a small team working on our ideas to learn things. For the logic and back-end side of our application, we have a part guys working on different parts with different languages (Rust, Go, Kotlin) the architecture for the back-end application we choose is microservice architecture and we also going to have a mobile app in android and iOS.
My question is can we use a separate database and back-end logic and still use Remix on the front-end web and have all the benefits it has?

**Comment by kiliman** [comment]
Yes, your Remix loaders and actions can use fetch to call external APIs. That way you still get the benefits of Remix data handling. If you would prefer to have the clients call your APIs directly, then you can use React Router 6.4 (due out next week) that has the loader/action support of Remix but entirely client side.

**Comment by sergiodxa** [comment]
As @kiliman said you can fetch your API from Remix loaders and actions.

I work on a Remix app that does this, we have a Rails API that is fetched from Remix and an iOS mobile app and it works really well since I can aggregate multiple API endpoints in my loaders and send the minimum required data to the UI.

**Comment by machour** [comment]
I too have been using Remix as BFF for a Yii framework powered API, and it's been a blast.

The Remix Client <=> Remix Server <=> Yii API is a really powerful workflow, where the Remix Server part proved to be more useful than one may think.

https://remix.run/docs/en/v1/guides/bff
